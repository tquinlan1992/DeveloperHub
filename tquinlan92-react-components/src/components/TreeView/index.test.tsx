import * as React from 'react';
import { shallow, configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import { TreeView } from './';

function getAnyJestFn() {
    return jest.fn() as any;
}

configure({ adapter: new Adapter() });

const mockActions = {
    onTagSelectionChange: getAnyJestFn(),
};

it('TreeView should render when given valid props', () => {
    const props = {
        tags: [],
        selectedTags: [],
        onTagSelectionChange: mockActions.onTagSelectionChange,
    };
    const result = shallow(<TreeView {...props} />);
    expect(result).toMatchSnapshot();
});
