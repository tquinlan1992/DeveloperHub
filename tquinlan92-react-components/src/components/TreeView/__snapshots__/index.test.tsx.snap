// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeView should render when given valid props 1`] = `
<div>
  <WithStyles(Button)
    onClick={[Function]}
  >
    Test Button
  </WithStyles(Button)>
  <h1>
    Tree View
  </h1>
</div>
`;
